dask/dataframe/core.py

```diff
@@ -322,7 +322,7 @@ def __init__(self, dsk, name, meta, divisions):
                 f"{typename(type(meta))}"
             )
         self._meta = meta
-        self.divisions = tuple(divisions)
+        self._divisions = tuple(divisions)
```

Comment:
Do we want this coercion still?

Reply:
Hmm good point removing this will probably expose several areas in the codebase where we are passing a list divisions to the constructor; happy to change this to something like

```suggestion
        self.divisions = divisions
```

IIRC that caused some issues with `_divisions` not being defined yet but that shouldn't be too difficult to resolve

Reply:
If it's not an antipattern to use the setter, that is probably easiest yeah.

Reply:
After working through some of the failures resulting from removing this coercion, having some second thoughts on removing it; this would essentially mean that all functions that implicitly set divisions (such as `repartition`) would need to pass tuple divisions to avoid breakage.

IMO, it would be less restrictive to users to allow iterable divisions for APIs that set them implicitly, and just silently coerce them to tuples internally; then the setter would only come into play when a user (or API) attempts to explicitly set divisions on a constructed frame.

Reply:
Ok I can see how that would become a massive change. It is fine with me to start with this, and we can always change it later if we decide that it's preferable to be more explicit.